src/builtins/BuiltinTemporal.cpp

```diff
@@ -491,6 +496,239 @@ static Value builtinTemporalPlainDateTimeInLeapYear(ExecutionState& state, Value
+    }
+
+    if (!TemporalInstantObject::isValidEpochNanoseconds(argv[0])) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "epoch Nanoseconds is out of range");
+    }
+
+    auto timeZone = TemporalTimeZoneObject::toTemporalTimeZone(state, argv[1]).asObject()->asTemporalTimeZoneObject();
+    auto calendar = TemporalCalendarObject::toTemporalCalendarWithISODefault(state, argc > 2 ? argv[2] : Value()).asObject()->asTemporalCalendarObject();
+    return TemporalZonedDateTimeObject::createTemporalZonedDateTime(state, *argv[0].asBigInt(), timeZone, calendar, newTarget);
+}
+
+static Value builtinTemporalZonedDateTimeFrom(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    Value options = Temporal::getOptionsObject(state, argc > 1 ? argv[1] : Value());
+
+    if (argv[0].isObject() && argv[0].asObject()->isTemporalZonedDateTimeObject()) {
+        Temporal::toTemporalOverflow(state, options);
+        Temporal::toTemporalDisambiguation(state, options);
+        Temporal::toTemporalOffset(state, options, Value(state.context()->staticStrings().reject.string()));
+        auto item = argv[0].asObject()->asTemporalZonedDateTimeObject();
+        return TemporalZonedDateTimeObject::createTemporalZonedDateTime(state, *(item->getNanoseconds()), item->getTimeZone(), item->getCalendar());
+    }
+
+    return TemporalZonedDateTimeObject::toTemporalZonedDateTime(state, argv[0], options);
+}
+
+static Value builtinTemporalZonedDateTimeCompare(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    return Value(TemporalInstantObject::compareEpochNanoseconds(state, *TemporalZonedDateTimeObject::toTemporalZonedDateTime(state, argv[0]).asObject()->asTemporalZonedDateTimeObject()->getNanoseconds(), *TemporalZonedDateTimeObject::toTemporalZonedDateTime(state, argv[1]).asObject()->asTemporalZonedDateTimeObject()->getNanoseconds()));
+}
+
+static Value builtinTemporalZonedDateTimeCalendar(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    CHECK_TEMPORAL_ZONED_DATE_TIME(state, thisValue)
+
+    return thisValue.asObject()->asTemporalZonedDateTimeObject()->getCalendar();
+}
+
+static Value builtinTemporalZonedDateTimeTimeZone(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    CHECK_TEMPORAL_ZONED_DATE_TIME(state, thisValue)
+
+    return thisValue.asObject()->asTemporalZonedDateTimeObject()->getTimeZone();
+}
+
+static Value builtinTemporalZonedDateTimeYear(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    CHECK_TEMPORAL_ZONED_DATE_TIME(state, thisValue)
+
+    return TemporalCalendarObject::calendarYear(state, (Object*)thisValue.asObject()->asTemporalZonedDateTimeObject()->getCalendar(), thisValue.asObject()->asTemporalZonedDateTimeObject()->toTemporalPlainDateTime(state));
```

Comment:
Does `(Object*)` cast needed? Many similar ones below.

Reply:
Yes because the `getCalendar` method returns `TemporalCalendarObject*` and the `calendarYear` method expects `Obect*`

Reply:
Object is a base class for it? It should be automatically casted then.